### PR TITLE
Created Check View

### PR DESCRIPTION
**Pull request by nkoblai**

### CODE REVIEW

File: moninag/check/models.py [hunk 1]
```diff
@@ -4,24 +4,159 @@
...
+        Args:
+            id(int, optional): check id. Defaults to None.
+            start(int, optional): start position when getting all checks. Default to 0.
+            end(int, optional): end position when getting all checks. Default to 20.
+
+        Returns:
+           QuerySet<Check>: QuerySet of checks.
+        """
+
+        return cls.objects.all()[start:end]
```
Comment (hutsulyk) [inline moninag/check/models.py]: `id` argument never used.

File: moninag/check/views.py [hunk 2]
```diff
@@ -1 +1,159 @@
...
+
+        check_dict = json.loads(request.body.decode('utf-8'))
+
+        if not _validate_dict(check_dict):
+            json_response['error'] = 'Incorect JSON format.'
+            return JsonResponse(json_response, status=400)
+
+        if Check.create(**check_dict):
+            json_response['response'] = Check.create(**check_dict).to_dict()
+            return JsonResponse(json_response, status=201)
```
Comment (hutsulyk) [inline moninag/check/views.py]: Create twice the same object `Check.create(**check_dict)`

File: moninag/moninag/settings.py [hunk 3]
```diff
@@ -50,7 +50,6 @@
     'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
-    'django.middleware.csrf.CsrfViewMiddleware',
```
Comment (hutsulyk) [inline moninag/moninag/settings.py]: Check if sign in, sign up still work without `CsrfViewMiddleware`

File: moninag/check/models.py (hunk 1 above)
Comment (nkoblai) [reply moninag/check/models.py]: id argument was deleted `def get(cls, start=0, end=9):`

File: moninag/check/views.py (hunk 2 above)
Comment (nkoblai) [reply moninag/check/views.py]: Using var `check` instead of.

File: moninag/moninag/settings.py (hunk 3 above)
Comment (nkoblai) [reply moninag/moninag/settings.py]: Added back.

File: moninag/moninag/urls.py [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 urlpatterns = [
     url(r'^', include('home.urls')),
+    url(r'^api_01/check/', include('check.urls')),
```
Comment (hutsulyk) [inline moninag/moninag/urls.py]: Replace on `url(r'^api/1/check/', include('check.urls')),`
I have changed too for Service View.
Liubomyr suggested :slightly_smiling_face:

Comment (nkoblai) [reply moninag/moninag/urls.py]: Done :sweat_smile:

**Comment by hutsulyk** [comment]
Update with dev :slightly_smiling_face:

File: moninag/check/models.py [hunk 5]
```diff
@@ -4,24 +4,158 @@
...
+            target_port(int): target port.
+            run_freq(int): run frequency.
+            service_id(int): service id.
+
+        Returns:
+            Check: The return value. Created check for success, None otherwise.
+        """
+
+        try:
+            service = Service.objects.get(id=service_id)
```
Comment (hutsulyk) [inline moninag/check/models.py]: Use `Service.get_by_id` insted.

File: moninag/check/models.py [hunk 6]
```diff
@@ -4,24 +4,158 @@
...
+
+        Returns:
+            Check: The return value. Updated check for success, None otherwise.
+        """
+
+        if cls.objects.filter(id=id).count() == 0:
+            return None
+
+        try:
+            service = Service.objects.get(id=service_id)
```
Comment (hutsulyk) [inline moninag/check/models.py]: Use `Service.get_by_id` insted.

File: moninag/check/views.py [hunk 7]
```diff
@@ -1 +1,161 @@
...
+        bool: The return value. True if valid, False otherwise.
+    """
+
+    REQUIREMENTS = {'name',
+                    'plugin_name',
+                    'run_freq',
+                    'target_port',
+                    'service_id'}
+
+    return not bool(REQUIREMENTS ^ set(dictionary.keys()))
```
Comment (hutsulyk) [inline moninag/check/views.py]: Use `validate_dict` from utils :slightly_smiling_face:

Comment (nkoblai) [reply moninag/check/views.py]: `validate_dict` imported from `utils`

File: moninag/check/models.py (hunk 5 above)
Comment (nkoblai) [reply moninag/check/models.py]: `get_by_id` is not a method of `Service` class

Comment (nkoblai) [reply moninag/check/models.py]: sry, my fault. Didnt rebase my local repo.

**Review by ghost — APPROVED** [review]

File: moninag/moninag/urls.py [hunk 8]
```diff
@@ -18,7 +18,8 @@
 
 
 urlpatterns = [
-    url(r'^', include('home.urls')),
+    url(r'^', include('home.urls')),w
```
Comment (hutsulyk) [inline moninag/moninag/urls.py]: What for `w`? :upside_down_face:

File: moninag/utils/validate.py [hunk 9]
```diff
@@ -0,0 +1,10 @@
+def validate_dict(dictionary, requirements):
+    """Validate given dictionary for required keys.
+    Args:
+        dictionary(dict): Dictionary to validate.
+        requirements(set): Set of required keys in dictionary.
+    Returns:
+        bool: The return value. True if valid, False otherwise.
+    """
+
+    return not bool(requirements ^ set(dictionary.keys()))
```
Comment (hutsulyk) [inline moninag/utils/validate.py]: Duplicated code. Use validator.py

File: moninag/check/models.py (hunk 5 above)
Comment (nkoblai) [reply moninag/check/models.py]: `Service.get_by_id` is working now.

File: moninag/check/models.py (hunk 6 above)
Comment (nkoblai) [reply moninag/check/models.py]: Replaced `Service.objects.get` via `Service.get_by_id`.

File: moninag/moninag/urls.py (hunk 8 above)
Comment (nkoblai) [reply moninag/moninag/urls.py]: Deleted 'w' symbol.

File: moninag/utils/validate.py (hunk 9 above)
Comment (nkoblai) [reply moninag/utils/validate.py]: Renamed `validate.py` into `validator.py`. Views now import utils.validator.

**Review by hutsulyk — CHANGES_REQUESTED** [review]
You shouldn't rename `validators.py` to `validator.py`, because other modules use it with name `validators`.
After rename they will not work. And they are not, try to runsever :upside_down_face: 

From the very beginning you had to use `validate_dict` function from already written `validators.py`.
There was no need to create your own file or modify existing.

**Comment by hutsulyk** [comment]
`urls.py` is missing for check.

File: moninag/check/models.py [hunk 10]
```diff
@@ -4,24 +4,158 @@
...
+            service_id(int): service id.
+
+        Returns:
+            Check: The return value. Created check for success, None otherwise.
+        """
+
+        try:
+            service = Service.get_by_id(id=service_id)
+        except:
+            return None
```
Comment (hutsulyk) [inline moninag/check/models.py]: `Service.get_by_id(id=service_id)` return None without exception, if service wasn't found.

File: moninag/check/models.py [hunk 11]
```diff
@@ -4,24 +4,158 @@
...
+            Check: The return value. Updated check for success, None otherwise.
+        """
+
+        if cls.objects.filter(id=id).count() == 0:
+            return None
+
+        try:
+            service = Service.get_by_id(id=service_id)
+        except:
+            return None
```
Comment (hutsulyk) [inline moninag/check/models.py]: `Service.get_by_id(id=service_id)` return None without exception, if service wasn't found.

**Review by hutsulyk — APPROVED** [review]

File: moninag/check/models.py [hunk 12]
```diff
@@ -4,24 +4,158 @@
...
     """
 
     name = models.CharField(max_length=20, default=0)
     plugin_name = models.CharField(max_length=20, default=0)
     target_port = models.IntegerField(default=0)
     run_freq = models.IntegerField(default=0)
-    service = models.ForeignKey(Service)
+    service = models.ForeignKey(Service, on_delete=models.CASCADE, default=0)
+
+    @classmethod
```
Comment (lhalam) [inline moninag/check/models.py]: use @staticmetod

File: moninag/check/models.py [hunk 13]
```diff
@@ -4,24 +4,158 @@
...
+            plugin_name(str): plugin name.
+            target_port(int): target port.
+            run_freq(int): run frequency.
+            service_id(int): service id.
+
+        Returns:
+            Check: The return value. Created check for success, None otherwise.
+        """
+
+        service = Service.get_by_id(id=service_id)
```
Comment (lhalam) [inline moninag/check/models.py]: Remove this line? because the method receives the existing service_id

File: moninag/check/models.py [hunk 14]
```diff
@@ -4,24 +4,158 @@
...
+                        target_port=target_port,
+                        run_freq=run_freq,
+                        service=service)
+            check.save()
+        else:
+            return None
+
+        return check
+
+    @classmethod
```
Comment (lhalam) [inline moninag/check/models.py]: remove @classmethod
it is instance method

**Review by lhalam — COMMENTED** [review]
make changes to files based on the comments

File: moninag/check/models.py [hunk 15]
```diff
@@ -4,24 +4,126 @@
...
+            Check: The return value. Created check for success, None otherwise.
+        """
+
+        check = Check()
+
+        check.name = name
+        check.plugin_name = plugin_name
+        check.target_port = target_port
+        check.run_freq = run_freq
+        check.service = Service.get_by_id(service_id)
```
Comment (hutsulyk) [inline moninag/check/models.py]: Except service instance as parameters instead of `service_id`

File: moninag/check/models.py [hunk 16]
```diff
@@ -4,24 +4,126 @@
...
+        if plugin_name:
+            self.plugin_name = plugin_name
+        if target_port:
+            self.target_port = target_port
+        if run_freq:
+            self.run_freq = run_freq
+
+        self.save()
+
+        return self
```
Comment (hutsulyk) [inline moninag/check/models.py]: Is it necessary to return self?

**Comment by hutsulyk** [comment]
Add user verification to view :slightly_smiling_face:

File: moninag/check/views.py [hunk 17]
```diff
@@ -1 +1,151 @@
...
+            check = Check.get_by_id(check_id)
+
+            if check:
+                json_response['response'] = check.to_dict()
+                return JsonResponse(json_response, status=200)
+            else:
+                json_response['error'] = 'Check with specified id was not found.'
+                return JsonResponse(json_response, status=404)
+
+        checks = Check.get()
```
Comment (ghost) [inline moninag/check/views.py]: 'Check' has no method 'get'

**Review by ghost — CHANGES_REQUESTED** [review]

File: moninag/moninag/urls.py [hunk 18]
```diff
@@ -20,6 +20,7 @@
 urlpatterns = [
     url(r'^', include('home.urls')),
     url(r'^api/1/service/', include('service.urls')),
+    url(r'^api/1/check/', include('check.urls')),
     # Registration & Auth
     url(r'^accounts/', include('registration.urls'))
 ]
```
Comment (hutsulyk) [inline moninag/moninag/urls.py]: Did you accidentally delete server api url? :slightly_smiling_face: 
`url(r'^api/1/server/', include('server.urls')),`

Comment (nkoblai) [reply moninag/moninag/urls.py]: Emmm, sry)

Comment (hutsulyk) [reply moninag/moninag/urls.py]: The root of problem is that your branch is not up-to-date with dev.

Rebase or merge with dev :upside_down_face:

**Comment by hutsulyk** [comment]
> This branch has conflicts that must be resolved

Update with dev :slightly_smiling_face:

**Comment by hutsulyk** [comment]
Update with dev again (`git rebase dev`) :slightly_smiling_face:

File: moninag/check/views.py [hunk 19]
```diff
@@ -1 +1,168 @@
...
+        :return: HttpResponse: Status 200 for success,
+                               Status 404 if not founded,
+                               Status 403 if Unauthorized.
+        """
+
+        check = Check.get_by_id(check_id)
+        if check.service.server.user.id == request.user.id:
+            count = check.delete()
+
+            if count:
```
Comment (lhalam) [inline moninag/check/views.py]: remove this check, it is not necessary

Comment (nkoblai) [reply moninag/check/views.py]: @lhalam removed check

**Review by hutsulyk — APPROVED** [review]

**Review by lhalam — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**